pkg/kgo/group_test.go

```diff
@@ -174,70 +174,75 @@ func (c *testConsumer) etl(etlsBeforeQuit int) {
 			}
 		}
 
-		// We poll with a short timeout so that we do not hang waiting
-		// at the end if another consumer hit the limit.
-		//
-		// If the host is slow, the context could be canceled immediately
-		// before we poll.
-		ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
-		fetches := cl.PollRecords(ctx, 100)
-		cancel()
-		if err := fetches.Err(); err == context.DeadlineExceeded || err == context.Canceled || err == ErrClientClosed {
-			if consumed := int(c.consumed.Load()); consumed == testRecordLimit {
-				return
-			} else if consumed > testRecordLimit {
-				panic(fmt.Sprintf("invalid: consumed too much from %s (group %s)", c.consumeFrom, c.group))
+		for range 4 {
```

Comment:
Please add doc comment that we loop four times to trigger repeated poll behavior while rebalancing is blocked.